### PR TITLE
Fix jaeger receiver to honor TLS config

### PR DESCRIPTION
**Pull request by dstdfx**
**Description:** 
Add missing TLSConfig into the jaeger receiver's configuration.
Create TLS listener if TLS config is passed.

**Link to tracking Issue:** #2850

### CODE REVIEW

**Comment by keitwb** [comment]
The `httpconfig.HTTPServerConfig` used has a `ToServer` and `ToListener` method that ensures all the config is used: https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/confighttp/confighttp.go#L158.

Is it possible to fit that into this receiver?

**Comment by dstdfx** [comment]
@keitwb I think it also makes sense to use `configgrpc.GRPCServerSettings` for GRPC collector and use available methods for building `net.Listener` and server options but it's a matter of another PR, what do you think?

**Comment by keitwb** [comment]
> @keitwb I think it also makes sense to use `configgrpc.GRPCServerSettings` for GRPC collector and use available methods for building `net.Listener` and server options but it's a matter of another PR, what do you think?

Yeah I agree, a different PR.

**Comment by keitwb** [comment]
@tigrannajaryan can you take a look?

**Review by jpkrohling — COMMENTED** [review]
The code itself looks good to me, but haven't there been any changes to the docs?

**Comment by dstdfx** [comment]
> The code itself looks good to me, but haven't there been any changes to the docs?

@jpkrohling As I see TLS configuration is already described here: 
https://github.com/open-telemetry/opentelemetry-collector/blob/main/receiver/jaegerreceiver/README.md#advanced-configuration

**Review by jpkrohling — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -55,6 +55,7 @@
 - `jaeger` receiver: Escape user input used in output (#2815)
 - `prometheus` exporter: Ensure same time is used for updated time (#2745)
 - `prometheusremotewrite` exporter: Close HTTP response body (#2875)
+- Fix Jaeger receiver to honor TLS Settings (#2866)
```
Comment (bogdandrutu) [inline CHANGELOG.md:58]: Please add this in the "unreleased" section.

Comment (dstdfx) [reply CHANGELOG.md]: Done

**Pull request merged**